src/Stratis.Bitcoin.Features.Consensus/PosConsensusFeature.cs

```diff
@@ -70,7 +70,7 @@ public override Task InitializeAsync()
             NetworkPeerConnectionParameters connectionParameters = this.connectionManager.Parameters;
 
             // Replace CMB.
-            bool oldCMBRemoved = connectionParameters.TemplateBehaviors.Remove(connectionParameters.TemplateBehaviors.Single(x => x is ConsensusManagerBehavior));
+            bool oldCMBRemoved = connectionParameters.TemplateBehaviors.Remove(connectionParameters.TemplateBehaviors.FirstOrDefault(x => x is ConsensusManagerBehavior));
```

Comment:
I think we should maybe stop the node if we don't have a `ConsensusManagerBehavior` as we should always have one. In the BaseFeature only, 5 of then are initialized and you can't run a node without the base feature.

So maybe check oldCMBRemoved is null and then throw an exception.

Reply:
actually it does the same, if null it will throw on the guard